mbtools/copy_resources_s3.py

```diff
@@ -0,0 +1,132 @@
+    for filename in os.listdir(resource_dir):
+
+        sha1 = hashlib.sha1()
+        full_path = os.path.join(resource_dir, filename)
+        with open(full_path, 'rb') as f:
+            while True:
+                data = f.read(buff_size)
+                if not data:
+                    break
+                sha1.update(data)
+        sha1_map[sha1.hexdigest()] = full_path
+    return sha1_map
+
+
+def get_mime_type(filename):
+    """Get the MIME type of file with libmagic """
+    mime_type = ''
+    try:
+        mime_type = magic.from_file(filename, mime=True)
+    finally:
+        return mime_type
+
+
+def add_new_resources_to_s3(bucket, s3_dir, hashes, hash2filename_map,
+                            metadata_file
+                            ):
+    """Add the files specified in filename_map to s3 iff they their
+    corresponding sha1 key exists in hashes"""
+
+    s3_client = boto3.client("s3")
+    hashkey2mimes_map = {}
+    hashkey2filenames_map = {}
+    hashkey2keypath_map = {}
+    for hash_key in hashes:
+        full_keypath = s3_dir + '/' + hash_key
+
+        try:
+            s3_client.head_object(Bucket=bucket,
+                                  Key=full_keypath)
+        except botocore.exceptions.ClientError:
+            mime_type = get_mime_type(hash2filename_map[hash_key])
+            s3_client.upload_file(hash2filename_map[hash_key],
+                                  Bucket=bucket,
+                                  Key=full_keypath,
+                                  ExtraArgs={
+                                    "ContentType": mime_type
+                                    }
+                                  )
+            hashkey2mimes_map[hash_key] = mime_type
+            hashkey2keypath_map[hash_key] = full_keypath
+            hashkey2filenames_map[hash_key] = os.path.basename(
+                                              hash2filename_map[hash_key]
+                                              )
+
+    add_metadata(hashkey2mimes_map,
+                 hashkey2filenames_map,
+                 hashkey2keypath_map,
+                 metadata_file)
+    print("SUCCESS")
+    print("Uploaded " + str(len(hashkey2mimes_map.keys())) + " files to " +
+          bucket)
+
+
+def add_metadata(new_tag_mimes, new_tag_filenames, new_tag_keypath,
+                 metadata_file
+                 ):
+    """Append metadata tags to a metadata_file"""
+
+    with open(metadata_file) as f:
+        data = json.load(f)
+        for key in new_tag_mimes:
+            tag = {"mime_type": new_tag_mimes[key],
+                   "sha1": key,
+                   "original_filename": new_tag_filenames[key],
+                   "s3_key": new_tag_keypath[key]
+                   }
+            data.append(tag)
+    with open(metadata_file, 'w') as f:
+        json.dump(data, f)
+
+
+def main():
+    new_resource_dir = Path(sys.argv[1]).resolve(strict=True)
+    metadata_file = Path(sys.argv[2]).resolve(strict=True)
```

Comment:
I think this got lost in my previous comment in this section of code, but this code assumes the metadata file exists already. It might be nice to create it if it doesn't to avoid having to manually create it.